### PR TITLE
Fix issues when querying from study view

### PR DESCRIPTION
**Pull request by kalletlak**

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -222,15 +219,58 @@
     window.cohortIdsList = ['hnsc_tcga'];
     window.iviz = {};
     window.iviz.datamanager = new DataManagerForIviz.init(window.cbioURL, studyCasesMap);
+    var getSelectableStudyIds = function() {
+      var def = new $.Deferred();
+      $.when(this.getPhysicalStudies(), this.getVirtualStudies()).then(function(physicalStudies, virtualStudies) {
+        var physicalStudyIds = _.pluck(physicalStudies,'studyId');
```
Comment (zhx828) [inline app/index.html]: `physicalStudyIds ` is not used anywhere?

File: app/index.html [hunk 2]
```diff
@@ -222,15 +219,58 @@
...
+          }
+        });
+        virtualStudyIds = _.flatten(virtualStudyIds);
+        def.resolve(virtualStudyIds)
+      }).fail(function(error) {
+        def.reject(error);
+      });
+      return def.promise();
+    }
+    var getVirtualStudies = function(){
```
Comment (zhx828) [inline app/index.html]: Just curious for future development, shouldn't this function access user email? Or the proxy endpoint will actually fetch studies for this user only?

File: app/index.html [hunk 3]
```diff
@@ -222,15 +219,58 @@
...
         if (_.isObject(configs)) {
           opts = configs;
         }
         $.when(
-          window.iviz.datamanager.initialSetup()
-        ).done(function(_data) {
+          window.iviz.datamanager.initialSetup(),
+          getSelectableStudyIds()
+        ).done(function(_data,selectableIds) {
           iViz.init(_data, opts);
```
Comment (zhx828) [inline app/index.html]: `selectableIds` hasn't been passed into `init` yet

File: app/scripts/main.js [hunk 4]
```diff
@@ -45,8 +46,20 @@ window.iViz = (function(_, $, cbio, QueryByGeneUtil, QueryByGeneTextArea) {
...
       vm_ = iViz.vue.manage.getInstance();
+      var selectableIdsSet = {}
+      _.each(_selectableIds, function(id){
+        selectableIdsSet[id] = true;
+      });
+
+      var cohortIds = window.cohortIdsList;
+      for (var i = 0; i < cohortIds.length; i++) {
+        if(selectableIdsSet[cohortIds[i]]){
+          includeCases = false;
```
Comment (zhx828) [inline app/scripts/main.js:59]: This will only be true is cohortIdsList is empty?

File: app/_components/QueryByGeneUtil.js [hunk 5]
```diff
@@ -24,76 +24,54 @@ window.QueryByGeneUtil = (function() {
...
+    * geneIds : selected genes
+    * includeCases : indicates whether to include custom samples in the form.
+    *                this is true for shared virtual study
+    */
+    query: function(cohortIds, stats, geneIds, includeCases) {
       var formOps = {
-        cancer_study_list: studyIds
-      };
+        cancer_study_list: cohortIds,
+        cancer_study_id:'all'
```
Comment (zhx828) [inline app/_components/QueryByGeneUtil.js:38]: In the original toMainPage function, cancer_study_id is not always `all`

File: app/_components/QueryByGeneUtil.js [hunk 6]
```diff
@@ -24,76 +24,54 @@ window.QueryByGeneUtil = (function() {
...
-
-        formOps = {
-          cancer_study_list: selectedCases.map(function(t) {
-            return t.id
-          }),
-          cancer_study_id: 'all',
-          case_set_id: -1,
-          case_ids: _arr.join('+')
-        };
+        formOps['case_set_id'] = -1;
```
Comment (zhx828) [inline app/_components/QueryByGeneUtil.js:68]: In original implementation, `case_set_id` is always -1

File: app/_components/QueryByGeneUtil.js [hunk 7]
```diff
@@ -24,76 +24,54 @@ window.QueryByGeneUtil = (function() {
...
-      submitForm(window.cbioURL + 'index.do', {
-        cancer_study_list: selectedCases.map(function(x) {
-          return x.id
-        }),
-        cancer_study_id: 'all',
-        case_ids: _arr.join('+'),
-        case_set_id: -1,
-        gene_set_choice: 'user-defined-list',
-        gene_list: encodeURIComponent(selectedGenes),
-        Z_SCORE_THRESHOLD: 2.0,
```
Comment (zhx828) [inline app/_components/QueryByGeneUtil.js:70]: This is gone?

File: app/_components/QueryByGeneUtil.js [hunk 8]
```diff
@@ -24,76 +24,54 @@ window.QueryByGeneUtil = (function() {
...
-        genetic_profile_ids_PROFILE_MUTATION_EXTENDED: mutationProfileId,
-        genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION: cnaProfileId,
-        clinical_param_selection: null,
```
Comment (zhx828) [inline app/_components/QueryByGeneUtil.js:73]: This is gone?

File: app/_components/QueryByGeneUtil.js [hunk 9]
```diff
@@ -24,76 +24,54 @@ window.QueryByGeneUtil = (function() {
...
-        });
-      });
-      submitForm(window.cbioURL + 'index.do', {
-        cancer_study_list: selectedCases.map(function(x) {
-          return x.id
-        }),
-        cancer_study_id: 'all',
-        case_ids: _arr.join('+'),
-        case_set_id: -1,
-        gene_set_choice: 'user-defined-list',
```
Comment (zhx828) [inline app/_components/QueryByGeneUtil.js:68]: This is gone?

**Review by zhx828 — CHANGES_REQUESTED** [review]
Need a second round review after questions/issues solved

File: app/_components/QueryByGeneUtil.js (hunk 5 above)
Comment (kalletlak) [reply app/_components/QueryByGeneUtil.js:38]: it should be ok setting it as `all` as the study ids would be present in `cancer_study_list`

File: app/_components/QueryByGeneUtil.js (hunk 6 above)
Comment (kalletlak) [reply app/_components/QueryByGeneUtil.js:68]: `case_set_id` is -1 only for custom case set

File: app/_components/QueryByGeneUtil.js (hunk 9 above)
Comment (kalletlak) [reply app/_components/QueryByGeneUtil.js:68]: I think by default it is 'user-defined-list'

File: app/_components/QueryByGeneUtil.js (hunk 7 above)
Comment (kalletlak) [reply app/_components/QueryByGeneUtil.js:70]: even this is by default  `2.0`

File: app/_components/QueryByGeneUtil.js (hunk 8 above)
Comment (kalletlak) [reply app/_components/QueryByGeneUtil.js:73]: I think this parameter is never used in back-end or front-end

File: app/index.html (hunk 2 above)
Comment (kalletlak) [reply app/index.html]: proxy api takes care of the email

File: app/scripts/main.js (hunk 4 above)
Comment (kalletlak) [reply app/scripts/main.js:59]: it will be true for shared VS

**Review by zhx828 — APPROVED** [review]

**Pull request merged**